### DISCUSSION TITLE
Best practice to save and restore Editor layout data.

### DISCUSSION THREAD

**Discussion by 99bits** [GrapesJS/grapesjs#5541] [Q&A]
Thank you for this amazing project.
We are new at GrapeJS and trying to make a kind of webpage builder where some custom components are being populated by dynamic data from API calls.  

Is there a best practice to being able to save and restore all of the grapejs components (both regular and custom components that gets its data dynamically from API call).
We are thinking of using Firebase as a backend.

Cheers.

**Comment by artf** [comment]
You can define a custom component that loads data from API and create inner components accordingly, those components are stored automatically inside the project JSON that you should simply decide where to save/load.
